File: Modul6/Bahasa_Python/PRAK604-2310817210012-Nazmi-Hakim.py
def koderahasia(pesan, kode):
    pesansimbol = [''] * 100
    bintang = 0
    pagar = 0
    panjang_pesan = len(pesan)
    panjang_kode = len(kode)

    if panjang_pesan != panjang_kode:
        print("Panjang kalimat berbeda, pesan palsu")
        return

    for i in range(panjang_pesan):
        if kode[i] == pesan[i]:
            pesansimbol[i] = ' ' if pesan[i] == ' ' else '*'
            bintang += pesan[i] != ' '
        else:
            pesansimbol[i] = '#'
            pagar += 1

    print(''.join(pesansimbol))
    print(f'* = {bintang}')
    print(f'# = {pagar}')
    if bintang >= pagar:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")

def main():
    pesan = input().rstrip()
    kode = input().rstrip()
    koderahasia(pesan, kode)

if __name__ == "__main__":
    main()

File: Modul6/Bahasa_C/PRAK604-2310817210012-Nazmi-Hakim.c
#include <stdio.h>
#include <string.h>

void koderahasia(char pesan[], char kode[]) {
    char pesansimbol[100];
    int bintang = 0, pagar = 0;
    int panjangPesan = strlen(pesan);
    int panjangKode = strlen(kode);

    if (panjangPesan != panjangKode) {
        printf("Panjang kalimat berbeda, pesan palsu\n");
        return;
    }

    for (int i = 0, panjangPesan = strlen(pesan); i < panjangPesan; i++) {
        if (kode[i] == pesan[i]) {
            pesansimbol[i] = (pesan[i] == ' ') ? ' ' : '*';
            bintang += (pesan[i] != ' ');
        } else {
            pesansimbol[i] = '#';
            pagar++;
        }
    }

    pesansimbol[strlen(pesan)] = '\0';
    printf("%s\n* = %d\n# = %d\n%s\n", pesansimbol, bintang, pagar,
           (bintang >= pagar) ? "Pesan Asli" : "Pesan Palsu");
}

int main() {
    char pesan[100], kode[100];
    fgets(pesan, sizeof(pesan), stdin);
    pesan[strcspn(pesan, "\n")] = '\0';
    fgets(kode, sizeof(kode), stdin);
    kode[strcspn(kode, "\n")] = '\0';
    koderahasia(pesan, kode);
    return 0;
}